### PR TITLE
fix(scripts): Update triggerShortcut.ts so that it works for non celo requests, comments on earn input

### PR DESCRIPTION
**Pull request by finnian0826**
triggerShortcut script didn't work out of the box for earn, had trouble parsing certain fields being passed in. Fixed how triggerInputShape is passed in.

Also updated chain used in wallet client to use the network ID rather than always using celo.

```console
yarn triggerShortcut --networkId arbitrum-one --address 0x644b896b0c45717215c0f10501b20fafd9bceb24 --app beefy --shortcut withdraw --triggerInputShape '{"tokens": [{"tokenId": "arbitrum-one:0xb9A27ba529634017b12e3cbbbFFb6dB7908a8C8B", "amount": "0.0000000000005"}], "tokenDecimals": 18, "positionAddress": "0xb9A27ba529634017b12e3cbbbFFb6dB7908a8C8B"}'
yarn run v1.22.21
$ ts-node ./scripts/triggerShortcut.ts --networkId arbitrum-one --address 0x644b896b0c45717215c0f10501b20fafd9bceb24 --app beefy --shortcut withdraw --triggerInputShape '{"tokens": [{"tokenId": "arbitrum-one:0xb9A27ba529634017b12e3cbbbFFb6dB7908a8C8B", "amount": "0.0000000000005"}], "tokenDecimals": 18, "positionAddress": "0xb9A27ba529634017b12e3cbbbFFb6dB7908a8C8B"}'
Triggering shortcut 'withdraw' for app 'beefy' {
  networkId: 'arbitrum-one',
  address: '0x644b896b0c45717215c0f10501b20fafd9bceb24',
  tokens: [
    {
      tokenId: 'arbitrum-one:0xb9A27ba529634017b12e3cbbbFFb6dB7908a8C8B',
      amount: '0.0000000000005'
    }
  ],
  positionAddress: '0xb9a27ba529634017b12e3cbbbffb6db7908a8c8b',
  tokenDecimals: 18
}
Transaction(s) to send: {
  transactions: [
    {
      networkId: 'arbitrum-one',
      from: '0x644b896b0c45717215c0f10501b20fafd9bceb24',
      to: '0xb9a27ba529634017b12e3cbbbffb6db7908a8c8b',
      data: '0x2e1a7d4d000000000000000000000000000000000000000000000000000000000007a120'
    }
  ]
}
No mnemonic provided, skipping transaction sending
```

### CODE REVIEW

File: scripts/triggerShortcut.ts [hunk 1]
```diff
@@ -71,6 +99,10 @@ void (async () => {
   // or at least a list of the expected fields
   // This just throws a Zod error if the input is not valid
   const triggerInput = z.object(shortcut.triggerInputShape).parse(argv)
+  // For deposit/withdraw shortcut, tokens field won't parse correctly. Instead, pass it in as
+  // a string and change the above to: {...shortcut.triggerInputShape, tokens: z.string()}
+  // Then below under ...triggerInput, add tokens: JSON.parse(triggerInput.tokens)
+  // Same happens for swap-deposit shortcut, but with swapFromToken field
```
Comment (satish-ravi) [inline scripts/triggerShortcut.ts:105]: would be nice to get this working automatically, wonder if we could just pass all of triggerInputShape as a json string and then do a JSON.parse and validate against the schema. @jeanregisser ideas?

**Review by satish-ravi — COMMENTED** [review]
question on PR title, this isn't specific to earn right? This fixes any non celo triggerShortcut request right?

File: scripts/triggerShortcut.ts (hunk 1 above)
Comment (finnian0826) [reply scripts/triggerShortcut.ts]: updated in [c56a987](https://github.com/valora-inc/hooks/pull/593/commits/c56a987fc765c8a9dbb817ef9f7cb727fa5665f8), example using it in the description

File: scripts/triggerShortcut.ts [hunk 2]
```diff
@@ -3,19 +3,47 @@
...
+  polygonAmoy,
+  sepolia,
+} from 'viem/chains'
 import { getShortcuts } from '../src/runtime/getShortcuts'
 import { NetworkId } from '../src/types/networkId'
 import { z } from 'zod'
 
 const CELO_DERIVATION_PATH = "m/44'/52752'/0'/0/0"
 
+const NETWORK_ID_TO_CHAIN = {
```
Comment (jeanregisser) [inline scripts/triggerShortcut.ts:26]: This way TS will catch when we add a new network and forget to add it here:
```suggestion
const NETWORK_ID_TO_CHAIN: Record<NetworkId, Chain> = {
```

File: scripts/triggerShortcut.ts [hunk 3]
```diff
@@ -43,6 +71,11 @@ const argv = yargs(process.argv.slice(2))
       type: 'string',
       demandOption: true,
     },
+    triggerInputShape: {
+      describe: 'triggerInputShape for the shortcut',
+      type: 'string',
```
Comment (jeanregisser) [inline scripts/triggerShortcut.ts:76]:
```suggestion
    triggerInput: {
      describe: 'JSON for the triggerInput of shortcut. It should respect the triggerInputShape of the shortcut.',
      type: 'string',
```

File: scripts/triggerShortcut.e2e.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import * as $ from 'shelljs'
 describe('triggerShortcut', () => {
   it('should trigger a shortcut successfully', () => {
     const result = $.exec(
-      'yarn triggerShortcut --network celo --address 0x2b8441ef13333ffa955c9ea5ab5b3692da95260d --app ubeswap --shortcut claim-reward --positionAddress 0xda7f463c27ec862cfbf2369f3f74c364d050d93f',
+      'yarn triggerShortcut --network celo --address 0x2b8441ef13333ffa955c9ea5ab5b3692da95260d --app ubeswap --shortcut claim-reward --triggerInputShape \'{"positionAddress": "0xda7f463c27ec862cfbf2369f3f74c364d050d93f"}\'',
```
Comment (jeanregisser) [inline scripts/triggerShortcut.e2e.ts:6]: nit: slightly easier to read
```suggestion
      `yarn triggerShortcut --network celo --address 0x2b8441ef13333ffa955c9ea5ab5b3692da95260d --app ubeswap --shortcut claim-reward --triggerInput '{"positionAddress": "0xda7f463c27ec862cfbf2369f3f74c364d050d93f"}'`,
```

**Review by jeanregisser — APPROVED** [review]
Thanks for fixing/improving it! 🚀 

Approved with some small suggestions.

File: scripts/triggerShortcut.ts (hunk 1 above)
Comment (jeanregisser) [reply scripts/triggerShortcut.ts]: Good idea @satish-ravi 👍

**Pull request merged**